packages/roosterjs-editor-dom/lib/edit/adjustInsertPosition.ts

```diff
@@ -160,28 +162,12 @@ function adjustInsertPositionForStructuredNode(
                     : PositionType.After
             );
         }
-    } else if (tag == 'TABLE' && trNode) {
-        // When inserting a table into a table, if these tables have the same column count, and
-        // current position is at beginning of a row, then merge these two tables
-        let newTable = new VTable(<HTMLTableElement>rootNodeToInsert);
-        let currentTable = new VTable(<HTMLTableCellElement>tdNode);
+    }
 
-        if (
-            currentTable.col == 0 &&
-            tdNode == currentTable.getCell(currentTable.row || 0, 0).td &&
-            newTable.cells?.[0] &&
-            newTable.cells[0].length == currentTable.cells?.[0].length &&
-            isPositionAtBeginningOf(position, tdNode)
-        ) {
-            if (
-                getTagOfNode(rootNodeToInsert!.firstChild) == 'TBODY' &&
-                !rootNodeToInsert!.firstChild?.nextSibling
-            ) {
-                unwrap(rootNodeToInsert!.firstChild!);
-            }
-            unwrap(rootNodeToInsert!);
-            position = new Position(trNode, PositionType.After);
```

Comment:
I think we still need these logic if isFragment is false, otherwise it will regress the existing behavior of insert table

Reply:
Try merge it into the new code you added to avoid duplicated code.

Reply:
Can you explain how it regresses insert table? It passes al tests. The removed code is already covered with the new paste table.

Reply:
Ok, for now it may not be a problem. But when doing part 2 and change insertTable to not using a fragment, it will be a problem. I'm ok to fix it together in part 2

Reply:
But what if I insert another table that has less column count than the existing one?